### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #77: Add a contributing guide
### Description

Add instructions on how to contribute.

### A/C
- CONTRIBUTING.md file at the root of the repo.

### PR DESCRIPTION
**Pull request by piotrpdev**
## Description

Adds a guide for contributing to the root of the repo; Port of the "Contributing Guide" section from the internal onboarding doc (will likely remove it once this is merged).

Resolves #77 

## How Has This Been Tested?

- **Visual Inspection**: Conducted a rigorous visual inspection using the latest in human optical technology to ensure the markdown file meets the highest standards of quality.

- **Scroll Test**: Engaged in a strenuous scroll test to confirm the document's scrollability on various devices, passing with flying colors.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: CONTRIBUTING.md
Comment (piotrpdev) [inline CONTRIBUTING.md:146]: I will update this section once/if #76 is merged

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:3]:
```suggestion
The following [`opendatahub-io/ai-edge`  GitHub Project board](https://github.com/orgs/opendatahub-io/projects/47) is the source of truth for the work taking place in this repo. You can pick up an issue from the TO DO column and follow the workflow described below. 
```

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:37]:
```suggestion
- If there is a roadblock to completing the issue, reach out on the relevant OpenDataHub Slack channels ([invite](https://github.com/opendatahub-io/opendatahub-community)). Someone will gladly try to help. 
```

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:45]:
```suggestion
    - Rebasing and squashing can be tricky so be careful while performing it. You can also learn more about [squashing commits with rebase](https://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).
```

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:46]:
```suggestion
  - Create the pull request, apply the appropriate labels, and link the relevant GitHub issue to it. Also make sure to include a good description of your changes and screenshots if appropriate.
```

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:44]:
```suggestion
  - When you’re ready to make a pull request with your changes, first clean-up by rebasing and squashing the commits. Make sure to use clear and descriptive commit messages.
```

File: CONTRIBUTING.md
Comment (devguyio) [inline CONTRIBUTING.md:48]:
```suggestion
  - Once any conflicts and code suggestions have been resolved, and your work has been approved, you can merge the pull request or wait for somebody else to merge it for you.
```

**Review by devguyio — CHANGES_REQUESTED** [review]
Thanks for doing this! I left a few comments

**Comment by piotrpdev** [comment]
> Thanks for doing this! I left a few comments

Thanks for the review :smile:

**Review by devguyio — APPROVED** [review]

**Comment by piotrpdev** [comment]
Will merge this if no more requested changes come in by the end of the day (~6PM `GMT+1` for me)

**Pull request merged**
